### PR TITLE
247-feat: Change type of ESLint config to flat-config

### LINKED ISSUE #247: Change type of ESLint config and add ESLint Stylistic
## 📝 Describe the feature

To make it easier to switch to the latest version of ESLint in the future, we are changing the type of ESLint config to `flat config` and adding ESLint Stylistic to maintain quality control of the code style at the highest level.

## ℹ️ Additional context

This task is subtask of #101 

## ✅ Definition of Done (DoD):

- change type of ESLint config to `flat config`;
- add ESLint Stylistic to our packages;
- add rules for stricter code style;
- check that everything is working correctly.

### PR DESCRIPTION
**Pull request by ansivgit**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [x] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description

Legacy ESLint config was changed to modern flat-config as part of the task of migrating to 9.x version of ESLint. In this PR:

- `eslintrc.js` config was changed to flat-config `eslint.config.js`;
- `ESLint Stylistic` with its `recommended` configuration was added for code style checking;
- Some basic configuration rules have been changed or additional rules have been added for more quality code style;
- `eslint-plugin-react` was added because it turned out that we don't track the quality of React code. This plugin was added in `package.json` and `eslint.config.js`, but now it doesn't work because we have too much issues about our code and fix it - out of scope this PR;
- some `ESLint Stylistic` issues in our code-base was fixed;
- `eslint-prettier-pluguin` was commented in `eslint.config.js` (not used for linting now) as it has conflicts with `ESLint Stylistic`. In addition, this linter has much more precise and flexible settings and I believe that we can abandon the `Prettier` altogether;
- Some installed `ESLint` plugins was updated to latest version.

## Related Tickets & Documents

- Related Issue #101 (as subtask)
- Related Issue #183 (as subtask)
- Closes #247 

## Additional info

For check this PR you need to remove `node_modules` and reinstall all packages.
Add in your IDE configuration the permission to use flat config
(e.g. for VS Code - `"eslint.experimental.useFlatConfig": true`).
This PR has `draft` status because we need to discuss some controversial points of the `Stylistic` configuration.

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

### CODE REVIEW

File: src/features/base-layout/components/footer/copyright.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @stylistic/jsx-one-expression-per-line */
```
Comment (natanchik) [inline src/features/base-layout/components/footer/copyright.tsx:1]: Do we need this?

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+build
+coverage
+dist
+
+optimizeImages
+src/__tests__
+
+# Ignore all HTML files:
+**/*.html
```
Comment (natanchik) [inline .prettierignore:12]: If we'll need it after Prettier's removal.

File: .prettierrc [hunk 3]
```diff
@@ -1,8 +1,5 @@
 {
-  "semi": true,
   "tabWidth": 2,
   "printWidth": 100,
-  "singleQuote": true,
-  "trailingComma": "all",
-  "jsxBracketSameLine": true
+  "singleQuote": true
 }
```
Comment (natanchik) [inline .prettierrc:5]: And this, maybe we should remove it?

File: src/app/components/date-lang/date-lang.tsx [hunk 4]
```diff
@@ -11,7 +12,9 @@ interface DateLangProps {
   type?: string;
 }
 
-export const DateLang = ({ startDate, language, mode, type = '' }: DateLangProps) => {
+export const DateLang = ({
+  startDate, language, mode, type = '',
+}: DateLangProps) => {
```
Comment (natanchik) [inline src/app/components/date-lang/date-lang.tsx:17]: I'm not sure which option is better

**Review by natanchik — APPROVED** [review]

File: src/features/base-layout/components/footer/copyright.tsx (hunk 1 above)
Comment (ansivgit) [reply src/features/base-layout/components/footer/copyright.tsx:1]: This is a temporary measure until we decide what we want to do with structures like this.

image

File: .prettierignore (hunk 2 above)
Comment (ansivgit) [reply .prettierignore:12]: Removed

File: .prettierrc (hunk 3 above)
Comment (ansivgit) [reply .prettierrc:5]: Removed

File: .prettierrc [hunk 5]
```diff
@@ -1,8 +1,5 @@
 {
-  "semi": true,
   "tabWidth": 2,
   "printWidth": 100,
-  "singleQuote": true,
-  "trailingComma": "all",
-  "jsxBracketSameLine": true
+  "singleQuote": true
```
Comment (andron13) [inline .prettierrc:4]: I thought we were giving up on prettierrc completely. Can you explain why we need it?

**Review by andron13 — COMMENTED** [review]
I agree in principle.

File: .prettierrc (hunk 5 above)
Comment (ansivgit) [reply .prettierrc:4]: The Stylelint-prettier plugin that formats our CSS depends on it. If we remove the config, Prettier requires double quotes and crashes the pipeline.

**Review by SpaNb4 — APPROVED** [review]

**Pull request merged**